src/it/java/io/deephaven/benchmark/tests/standard/parquet/ParquetTestRunner.java

```diff
@@ -0,0 +1,208 @@
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import io.deephaven.benchmark.api.Bench;
+import io.deephaven.benchmark.metric.Metrics;
+import io.deephaven.benchmark.util.Exec;
+import io.deephaven.benchmark.util.Timer;
+
+/**
+ * Test reading and writing parquet files with various data types and compression codecs.
+ */
+class ParquetTestRunner {
+    final Object testInst;
+    final Bench api;
+    private int rowCountFactor = 1;
+    private int scaleFactor = 1;
+    private long scaleRowCount;
+
+    ParquetTestRunner(Object testInst) {
+        this.testInst = testInst;
+        this.api = initialize(testInst);
+        this.scaleRowCount = api.propertyAsIntegral("scale.row.count", "100000");
+    }
+
+    /**
+     * Set a multiplier for the generated rows and a multiplier for simulating more rows with {@code merge}
+     * 
+     * @param rowCountFactor the multiplier for the scale.row.count property
+     * @param scaleFactor the multiplier for how many merges to do on the generated table to simulate more rows
+     */
+    void setScaleFactors(int rowCountFactor, int scaleFactor) {
+        this.rowCountFactor = rowCountFactor;
+        this.scaleRowCount = (long) (api.propertyAsIntegral("scale.row.count", "100000") * rowCountFactor);
+        this.scaleFactor = scaleFactor;
+    }
+
+    /**
+     * Read a benchmark that measures parquet read performance. This tests always runs after a corresponding write test.
+     * 
+     * @param testName name that will appear in the results as the benchmark name
+     */
+    void runReadTest(String testName) {
+        var q = """
+        bench_api_metrics_snapshot()
+        begin_time = time.perf_counter_ns()
+        source = read('/data/source.ptr.parquet').select()
+        end_time = time.perf_counter_ns()
+        bench_api_metrics_snapshot()
+        standard_metrics = bench_api_metrics_collect()
+        
+        stats = new_table([
+            double_col("elapsed_nanos", [end_time - begin_time]),
+            long_col("processed_row_count", [source.size]),
+            long_col("result_row_count", [source.size])
+        ])
+        """;
+        runTest(testName, q);
+    }
+
+    /**
+     * Run a benchmark the measures parquet write performance.
+     * 
+     * @param testName the benchmark name to record with the measurement
+     * @param codec a compression codec
+     * @param columnNames the names of the pre-defined columns to generate
+     */
+    void runWriteTest(String testName, String codec, String... columnNames) {
+        var q = """
+        source = merge([empty_table(${rowCount}).update([
+            ${generators}
+        ])] * ${scaleFactor})
+        
+        bench_api_metrics_snapshot()
+        begin_time = time.perf_counter_ns()
+        write(
+            source, '/data/source.ptr.parquet', compression_codec_name='${codec}', 
+            max_dictionary_keys=2000000, max_dictionary_size=20000000, target_page_size=2000000
```

Comment:
Yes, Ryan advised setting the sizes explicitly so that if we changed the defaults underneath, the tests would be unaffected.  Are you saying that you would like a test that uses the defaults, so the it gets flagged if the defaults change?

Reply:
That's actually a good idea. Because I am spending some effort on tweaking the defaults, and the impact for any change in the default should be established at the time of release.

Reply:
Added multi-column tests like the other multi-column ones that uses whatever defaults are in your parquet code.